src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/Workspace.cs

```diff
@@ -43,9 +43,6 @@ internal enum WorkspaceState
     private readonly CancellationToken _unloadCancellationToken;
     private readonly string _baseDirectory;
 
-    /// <summary>Completes when the workspace has integrated evaluation data.</summary>
-    private readonly TaskCompletionSource _hasEvaluationData = new(TaskCreationOptions.RunContinuationsAsynchronously);
```

Comment:
This class no longer has to track whether it has seen evaluation data. The responsibility of ensuring the first update contains evaluation data has moved to the caller. This class now throws if that expectation is not met.